### PR TITLE
Fix the segmentation fault for Ascend all, any with dim being None

### PR DESCRIPTION
**Pull request by jfxu-st**
## Motivation and Context

all, any on Ascend can cause a segmentation fault when dim is None. This is the root cause of the failure of many unit tests in DIPU, e.g., test_multinomial, test_all, test_any.

## Description

- Fix the segmentation fault for all, any.
- Add a new test case for the all, any ops where dim is None 
- Add skip for reduce_op_1 in device_configs.py for topsrider

## Checklist
**Before PR**:

- [x] I have read and followed the workflow indicated in the [Contributors.md](https://github.com/DeepLink-org/DIOPI/blob/main/Contributors.md) to create this PR.
- [x] Pre-commit or linting tools indicated in [Contributors.md](https://github.com/DeepLink-org/DIOPI/blob/main/Contributors.md) are used to fix the potential lint issues.
- [x] Bug fixes are covered by unit tests, the case that causes the bug should be added in the unit tests.
- [x] New functionalities are covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [x] The documentation has been modified accordingly, including docstring or example tutorials.

**After PR**:

- [x] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

**Review by zhangzefeng92 — APPROVED** [review]

File: impl/ascend/functions/reduce.cpp [hunk 1]
```diff
@@ -111,14 +111,24 @@ diopiError_t diopiMean(diopiContextHandle_t ctx, diopiTensorHandle_t out, diopiC
     return diopiSuccess;
 }
 
+inline std::vector<int64_t> getDimVectorForTensor(diopiConstTensorHandle_t th) {
+    AscendTensor at(th);
+    std::vector<int64_t> dimVector;
```
Comment (yangbofun) [inline impl/ascend/functions/reduce.cpp:116]: vector在赋值前，如果知道了有多大，就提前在构造时写好大小。

Comment (jfxu-st) [reply impl/ascend/functions/reduce.cpp]: > vector在赋值前，如果知道了有多大，就提前在构造时写好大小。

Done

**Review by yangbofun — APPROVED** [review]
